Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesInterface.h

```diff
@@ -347,6 +366,9 @@ namespace AzFramework
         {
             return reinterpret_cast<const T*>(ticket->m_payload);
         }
+
+        AZStd::unordered_map<EntitySpawnTicket::Id, EntitySpawnTicket*> m_entitySpawnTicketMap;
+        AZStd::mutex m_entitySpawnTicketMapMutex;
```

Comment:
While there's very little chance in this case for this function to recurse, I would still recommend using `AZStd::recursive_mutex` just to be sure. This recursive mutex lock issues won't show on Windows, but will happen on other platforms.